File: app/src/main/java/com/example/pozoleria/CategoriesActivity.java
package com.example.pozoleria;

import android.os.Bundle;
import androidx.appcompat.app.AppCompatActivity;
import androidx.recyclerview.widget.GridLayoutManager;
import androidx.recyclerview.widget.RecyclerView;

import java.util.ArrayList;
import java.util.List;

public class CategoriesActivity extends AppCompatActivity {

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_categories);

        // 1. Referencia al RecyclerView
        RecyclerView recycler = findViewById(R.id.recyclerCategoriasFull);

        // 2. Lista de categorías con imágenes
        List<CategoryItem> categorias = new ArrayList<>();
        categorias.add(new CategoryItem("Pozole", R.drawable.pozole));
        categorias.add(new CategoryItem("Tacos Dorados", R.drawable.tacos_dorados));
        categorias.add(new CategoryItem("Postres", R.drawable.postres));
        categorias.add(new CategoryItem("Bebidas", R.drawable.refresco));
        categorias.add(new CategoryItem("Combos", R.drawable.combos));
        categorias.add(new CategoryItem("Promociones", R.drawable.promociones));

        // 3. Configurar Adapter
        CategoryAdapter adapter = new CategoryAdapter(this, categorias);

        // 4. Layout tipo grid de 2 columnas
        recycler.setLayoutManager(new GridLayoutManager(this, 2));

        // 5. Asignar el adapter al RecyclerView
        recycler.setAdapter(adapter);
    }
}

File: app/src/main/java/com/example/frontend/CategoriesActivity.kt
package com.example.frontend

import android.content.Intent
import android.os.Bundle
import androidx.appcompat.app.AppCompatActivity
import androidx.recyclerview.widget.GridLayoutManager
import androidx.recyclerview.widget.RecyclerView
import com.example.frontend.adapters.CategoryAdapter
import com.example.frontend.models.Category
import com.google.android.material.floatingactionbutton.FloatingActionButton

class CategoriesActivity : AppCompatActivity() {

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_categories)

        val recycler = findViewById<RecyclerView>(R.id.recyclerCategories)
        recycler.layoutManager = GridLayoutManager(this, 2)

        val categories = listOf(
            Category("1", "Promociones", R.drawable.promociones),
            Category("2", "Platillos", R.drawable.desayunos),
            Category("3", "Postres", R.drawable.postres),
            Category("4", "Bebidas", R.drawable.bebidas),
            Category("5", "Combos", R.drawable.combos)
        )

        recycler.adapter = CategoryAdapter(categories) { category ->
            val intent = Intent(this, ProductsActivity::class.java)
            intent.putExtra("categoryName", category.name)
            startActivity(intent)
        }

        val fabCart = findViewById<FloatingActionButton>(R.id.fabCart)
        fabCart.setOnClickListener {
            val intent = Intent(this, CartActivity::class.java)
            startActivity(intent)
        }
    }
}
